### PR TITLE
Supported socks5 proxy for crypto-ws-client

### PR DESCRIPTION
**Pull request by GhostLee**
Use `fast_socks5` crate to wrap the TcpStream so that we can use socks5 proxy, please set https_proxy env before run executable.
eg. `https_proxy=socks5://127.0.0.1:10808`

### CODE REVIEW

File: crypto-ws-client/src/common/connect_async.rs [hunk 1]
```diff
@@ -17,6 +20,24 @@ pub async fn connect_async(
     url: &str,
     uplink_limit: Option<(NonZeroU32, std::time::Duration)>,
 ) -> Result<(Receiver<Message>, Sender<Message>), Error> {
+
+    if let Ok(proxy_env) = env::var("https_proxy") {
```
Comment (soulmachine) [inline crypto-ws-client/src/common/connect_async.rs:24]: Please also support `http_proxy`, for example, `if let Ok(proxy_env) = env::var("https_proxy").or(env::var("http_proxy"))`

**Comment by soulmachine** [comment]
Hi @GhostLee  , thank your for this great job. One quick question,  have you run this against a real socks5 proxy? Is your code working?

**Comment by soulmachine** [comment]
Related to https://github.com/snapview/tungstenite-rs/issues/177

**Comment by GhostLee** [comment]
yes, I use it in my project.

**Review by soulmachine — APPROVED** [review]
Great job!

**Comment by soulmachine** [comment]
I published a  new versions of `carbonbot` and `crypto-crawler`, which  support socks proxy now.

**Pull request merged**